chain/client/src/client_actor.rs

```diff
@@ -118,7 +119,12 @@ pub struct ClientActor {
 
     /// Synchronization measure to allow graceful shutdown.
     /// Informs the system when a ClientActor gets dropped.
-    shutdown_signal: Option<oneshot::Sender<()>>,
+    shutdown_signal: Option<broadcast::Sender<()>>,
+
+    dyn_config: Arc<Mutex<Option<DynConfig>>>,
```

Comment:
I don't think the `Arc` or the `Mutex` are necessary here. `dyn_config` is only owned by `ClientActor`, so you can just make it `Option<DynConfig>` and simply update the field

Reply:
Done